pkg/manifest/mantaray/marshal_test.go

```diff
@@ -201,3 +202,58 @@ func TestMarshal(t *testing.T) {
+			err:  ErrInvalidVersionHash,
+		},
+		{ // fork with metadata, where metadata size value is less then metadata content (size 89, want 93)
+			data: decode("00000000000000000000000000000000000000000000000000000000000000005768b3b6a7db56d21d1abff40d41cebfc83448fed8d7e9b06ec0d3b073f28f200000000000000000000000000000000000000000000000000000000000000000000000000080000000000000000000000000000000000000000000000000000016012f0000000000000000000000000000000000000000000000000000000000e87f95c3d081c4fede769b6c69e27b435e525cbd25c6715c607e7c531e32963900597b22776562736974652d696e6465782d646f63756d656e74223a2233356561656538316262363338303436393965633637316265323736326465626665346662643330636461646139303232393239646131613965366134366436227d0a"),
+			err:  ErrInvalidOutput,
+		},
+		{ // fork with vaild metadata (size is 93)
+			data: decode("00000000000000000000000000000000000000000000000000000000000000005768b3b6a7db56d21d1abff40d41cebfc83448fed8d7e9b06ec0d3b073f28f200000000000000000000000000000000000000000000000000000000000000000000000000080000000000000000000000000000000000000000000000000000016012f0000000000000000000000000000000000000000000000000000000000e87f95c3d081c4fede769b6c69e27b435e525cbd25c6715c607e7c531e329639005d7b22776562736974652d696e6465782d646f63756d656e74223a2233356561656538316262363338303436393965633637316265323736326465626665346662643330636461646139303232393239646131613965366134366436227d0a"),
+			err:  nil,
+		},
+		{ // fork with metadata, where metadata size value exceeds actual metadata size (size is 95, want 93)
+			data: decode("00000000000000000000000000000000000000000000000000000000000000005768b3b6a7db56d21d1abff40d41cebfc83448fed8d7e9b06ec0d3b073f28f200000000000000000000000000000000000000000000000000000000000000000000000000080000000000000000000000000000000000000000000000000000016012f0000000000000000000000000000000000000000000000000000000000e87f95c3d081c4fede769b6c69e27b435e525cbd25c6715c607e7c531e329639005f7b22776562736974652d696e6465782d646f63756d656e74223a2233356561656538316262363338303436393965633637316265323736326465626665346662643330636461646139303232393239646131613965366134366436227d0a"),
+			err:  ErrInvalidOutput,
+		},
+		{ // fork with metadata, where metadata size value exceeds actual metadata size (size is 96, want 93)
+			data: decode("00000000000000000000000000000000000000000000000000000000000000005768b3b6a7db56d21d1abff40d41cebfc83448fed8d7e9b06ec0d3b073f28f200000000000000000000000000000000000000000000000000000000000000000000000000080000000000000000000000000000000000000000000000000000016012f0000000000000000000000000000000000000000000000000000000000e87f95c3d081c4fede769b6c69e27b435e525cbd25c6715c607e7c531e32963900607b22776562736974652d696e6465782d646f63756d656e74223a2233356561656538316262363338303436393965633637316265323736326465626665346662643330636461646139303232393239646131613965366134366436227d0a"),
+			err:  ErrInvalidOutput,
+		},
+	}
+
+	for _, tc := range tests {
+		n := New()
+		err := n.UnmarshalBinary(tc.data)
```

Comment:
Naming it `haveErr` or `gotErr` would be more indicative.

Reply:
I my option `err` is equally indicative here and both suggestions seem verbose. Variable name `err` is usual convention used to name error result from last function calls. 

However I have updated `haveErr`.